Fix the bug where multiple displays of a video affect its playback rate

sukibaby opened the pull request:
Closes #576

This resolves the issue where a video being displayed more than once results in the playback rate of the video being affected relative to the number of times it is shown. 

As an example this might be if  the same video is used as a background and a banner, and the player has Step Statistics enabled

teejusb commented:
Hmm this seems like it would load the texture into memory again, which essentially defeats the purpose of the texture manager (which is to save memory for duplicate textures). Videos are also the largest in size so this impacts memory the most.

That being said in this current day and age, videos aren't all that common yet so I'm actually okay with this. It doesn't actually answer the question as to what causes access to the same video to play back at double the rate -- so it's currently masking that issue. I would add a comment describing that this is a workaround for that behavior and that more work is needed to investigate the root cause. Since this fixes the issue from a general player's POV I'm happy to merge this though.

sukibaby commented:
> It doesn't actually answer the question as to what causes access to the same video to play back at double the rate 

We're apparently unable to share the texture itself without also sharing state

sukibaby committed:
Ensure unique instances for movie textures

Adds a workaround to prevent reuse of movie textures by making their IDs unique per load. This fixes a bug where movie textures and their properties were unintentionally shared when the same video was displayed multiple times, such as in backgrounds and banners.
